### PR TITLE
Improve documentation around static evaluation

### LINKED ISSUE #1833: Docs: Update documentation for Modules, Backends and State Encryption, based on Static Evaluation support
[State Encryption](https://opentofu.org/docs/language/state/encryption/), [Backend Configuration](https://opentofu.org/docs/language/settings/backends/configuration/), [Cloud Configuration](https://opentofu.org/docs/language/settings/tf-cloud/) and [Module Sources](https://opentofu.org/docs/language/modules/sources/) should be updated to mention that variables and locals and interpolations are supported

Some notes:
- For [Backend Configuration](https://opentofu.org/docs/language/settings/backends/configuration/) - there's the [Partial Configuration](https://opentofu.org/docs/language/settings/backends/configuration/#partial-configuration) part. Should be updated to mention using variables and locals, and probably prefer this over the existing `-backend-config` flag
- Should contain a warning, saying that using variables or locals that rely on variables would require you to provide these variables in commands that would otherwise not need variable values

### PR DESCRIPTION
**Pull request by cam72cam**
Resolves #1833 and #1835

## Target Release

1.8.0, 1.9.0

## Checklist

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.

### Website/documentation checklist

- [x] I have locally started the website as [described here](https://github.com/opentofu/opentofu/blob/main/website/README.md) and checked my changes.

### CODE REVIEW

File: website/docs/cli/commands/console.mdx [hunk 1]
```diff
@@ -9,6 +9,14 @@ description: >-
 The `tofu console` command provides an interactive console for
 evaluating [expressions](../../language/expressions/index.mdx).
 
+:::note
```
Comment (Evi1Pumpkin) [inline website/docs/cli/commands/console.mdx:12]: Nitpick, but sometimes this note is before the "Usage" section, and sometimes it is in the "Usage" section right before the flags are mentioned.
Let's be consistent and choose a single place. I think it makes more sense to include this section inside the  "Usage" section, as it's relevant to how you use the command.

Comment (ollevche) [reply website/docs/cli/commands/console.mdx]: Yep, makes sense. Moved note to usage block for most of the commands: [3980be3](https://github.com/opentofu/opentofu/pull/1843/commits/3980be38e72560ca2a97aea35f51c0f0ea66b9cf)

**Review by Evi1Pumpkin — DISMISSED** [review]
LGTM!

File: website/docs/cli/commands/console.mdx (hunk 1 above)
Comment (ollevche) [reply website/docs/cli/commands/console.mdx]: that's a miss, thank you!

[e27f67c](https://github.com/opentofu/opentofu/pull/1843/commits/e27f67c2a6ed5485312283c102c84e11accd337b)

**Review by Evi1Pumpkin — APPROVED** [review]

**Pull request merged**